rte/kernels/mo_rte_solver_kernels.F90

```diff
@@ -947,5 +949,295 @@ pure subroutine apply_BC_0(ncol, nlay, ngpt, top_at_1, flux_dn) bind (C, name="a
+
+  do imu = 2, nmus
+    Ds_ncol(:,:) = Ds(imu)
+    weight = 2._wp*pi*weights(imu)
+    radn_dn(1:ncol, top_level, 1:ngpt)  = flux_dn(1:ncol, top_level, 1:ngpt) / weight
+    call lw_solver_1rescl(ncol, nlay, ngpt, &
+                          top_at_1, Ds_ncol, tauLoc, scaling, &
+                          lay_source, lev_source_inc, lev_source_dec, sfc_emis, sfc_src, &
+                          radn_up, radn_dn)
+
+    flux_up(:,:,:) = flux_up(:,:,:) + weight*radn_up(:,:,:)
+    flux_dn(:,:,:) = flux_dn(:,:,:) + weight*radn_dn(:,:,:)
+  end do
+  end subroutine lw_solver_1rescl_GaussQuad
+
+  pure subroutine scaling_1rescl(ncol, nlay, ngpt, tauLoc, scaling, tau, ssa, g)
+    integer ,                              intent(in)    :: ncol
+    integer ,                              intent(in)    :: nlay
+    integer ,                              intent(in)    :: ngpt
+    real(wp), dimension(ncol, nlay, ngpt), intent(in)    :: tau
+    real(wp), dimension(ncol, nlay, ngpt), intent(in)    :: ssa
+    real(wp), dimension(ncol, nlay, ngpt), intent(in)    :: g
+
+    real(wp), dimension(ncol, nlay, ngpt), intent(inout) :: tauLoc
+    real(wp), dimension(ncol, nlay, ngpt), intent(inout) :: scaling
+
+    integer  :: icol, ilay, igpt
+    real(wp) :: wb, ssal, scaleTau
+    !$acc enter data copyin(tau, ssa, g)
+    !$acc enter data create(tauLoc, scaling)
+    !$acc parallel loop collapse(3)
+    do igpt=1,ngpt
+      do ilay=1,nlay
+        do icol=1,ncol
+          ssal = ssa(icol, ilay, igpt)
+          wb = ssal*(1._wp - g(icol, ilay, igpt)) / 2._wp
+          scaleTau = (1._wp - ssal + wb )
+          ! Eq.15 of the paper
+          tauLoc(icol, ilay, igpt) = scaleTau * tau(icol, ilay, igpt)
+          ! 
+          ! here ssa is used to store parameter wb/[1-w(1-b)] of Eq.21 of the Tang's paper
+          ! actually it is in line of parameter rescaling defined in Eq.7
+          if (scaleTau > epsilon(1._wp)) then
+            scaling(icol, ilay, igpt) = wb / scaleTau
+          else
+            scaling(icol, ilay, igpt) = 1._wp
+          endif
+        enddo
+      enddo
+    enddo
+    !$acc exit data copyout(tauLoc, scaling)
+    !$acc exit data delete(tau, ssa, g)
+  end subroutine scaling_1rescl
+
+! -------------------------------------------------------------------------------------------------
+!
+! Longwave no-scattering + a single itergation transport
+!
+! 
+! Implemented based on the paper
+! Tang G, P Yang, GW Kattawar, X Huang, EJ Mlawer, BA Baum, MD King, 2018: Improvement of 
+! the Simulation of Cloud Longwave Scattering in Broadband Radiative Transfer Models, 
+! Journal of the Atmospheric Sciences 75 (7), 2217-2233
+! https://doi.org/10.1175/JAS-D-18-0014.1
+! adustment terms is computed based on solution of the Tang equations
+! for "linear-in-tau" internal source
+! 
+! -------------------------------------------------------------------------------------------------
+subroutine lw_transport_1rescl(ncol, nlay, top_at_1, &
```

Comment:
By requiring `An` and `Cn` as inputs you're expecting that users will have called `scaling_1rescl()`. This works in the current implementation, since all roads to `lw_solver_1rescl()` lead through `lw_solver_1rescl_GaussQuad()`, where you are computing the scaling. But when we introduce Jen's optimal angle this might not be true. I don't have a good answer but I wanted to flag the issue

Reply:
I added comment to lw_solver_1rescl saying that the user must provide the correct scaling to the call of the function